alchemy/src/railway/database.ts

```diff
@@ -0,0 +1,257 @@
+import type { Context } from "../context.ts";
+import { Resource } from "../resource.ts";
+import { type Secret, secret } from "../secret.ts";
+import { createRailwayApi, handleRailwayDeleteError } from "./api.ts";
+
+export interface DatabaseProps {
+  /**
+   * The name of the database
+   */
+  name: string;
+
+  /**
+   * The ID of the project this database belongs to
+   */
+  projectId: string;
```

Comment:
Whenever an input property references a domain specific entity, e.g project. We should model it like:

project: string | Project

That way our model is "lifted" into the alchemy abstraction while also supporting string references to externally defined resources. 

Please go ahead and apply this to all properties in the new railway resources.